src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy

```diff
@@ -595,6 +595,21 @@ class PipelineTestHelper {
         }.count()
     }
 
+    /**
+     * Verifies if a method was called, with the preconditions defined in times and methodVerification, if wanted.
+     * @param name the method name
+     * @param times times the method shall be called.
+     * @param methodVerification a closure with the a MethodSignature object as input parameter, which verifies a condition
+     */
+    void verify(String name, Integer times = 1, Closure methodVerification = {true}) {
+        List<MethodCall> methodCalls = callStack.findAll {it.getMethodName() == name }
+        if(!methodCalls.every(methodVerification))
+            throw new VerificationException("Not every call of $name passed the verifications: $methodCalls")
+        def timesCalled = methodCalls.size()
+        if(timesCalled != times)
+            throw new VerificationException("Method $name expected $times, but called $timesCalled")
+    }
```

Comment:
```suggestion
    void verify(String name, int times = 1, Closure methodVerification = { return true }) {
        List<MethodCall> methodCalls = callStack.findAll { it.getMethodName() == name }
        methodCalls.each { call ->
            if (!methodVerification(call)) {
                throw new VerificationException("Method call $call failed to be verified")
            }
        }
        int timesCalled = methodCalls.size()
        if (times != timesCalled) {
            throw new VerificationException("Expected method $name to be called $times times, but was called $timesCalled times")
        }
    }
```

Aside from a bit of grammar cleanups and formatting, I changed the verification loop so that it actually prints out the call that failed. However, this will of course stop after the first verification failure. If you'd prefer, you could also collect the failures in a list and log them all. But IMHO it's enough to just halt after one verification fails.

Reply:
Done